pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -3813,6 +3797,128 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 			Entry("explicit only", useExplicitHyperV, doNotUseHyperVPassthrough, true),
 			Entry("no hyperv use", doNotUseExplicitHyperV, doNotUseHyperVPassthrough, true),
 		)
+
+	})
+
+	Context("Watchdog device validation", func() {
+		DescribeTable("should validate watchdog devices per arch",
+			func(vmi *v1.VirtualMachineInstance, validator func(*k8sfield.Path, *v1.VirtualMachineInstanceSpec) []metav1.StatusCause, expectedMessage string, shouldReject bool) {
```

Comment:
I have 2 things here:
1. Instead of giving a vmi creation function, initialize the either in the test or in a `BeforeEach`. Then as parameter take watchdog device and maybe the architecture.
2. Please do not use the validator function as a variable but instead call a common entrypoint, to ensure that the logic choosing the right validator function is also tested.

Reply:
Made changes as mentioned